### PR TITLE
Return fields on project list

### PR DESCRIPTION
**Pull request by pdeziel**
### Scope of changes

This updates the tenant project list handler so that all of the required fields are returned.

Fixes SC-15720

### Type of change

- [x] new feature
- [ ] bug fix
- [ ] documentation
- [ ] testing
- [ ] technical debt
- [ ] other (describe)

### Acceptance criteria

Describe how reviewers can test this change to be sure that it works correctly. Add a checklist if possible.

### Author checklist

- [x] I have manually tested the change and/or added automation in the form of unit tests or integration tests
- [ ]  I have updated the dependencies list
- [ ]  I have recompiled and included new protocol buffers to reflect changes I made
- [ ]  I have added new test fixtures as needed to support added tests
- [x]  Check this box if a reviewer can merge this pull request after approval (leave it unchecked if you want to do it yourself)
- [x]  I have moved the associated Shortcut story to "Ready for Review"

### Reviewer(s) checklist

- [ ] Any new user-facing content that has been added for this PR has been QA'ed to ensure correct grammar, spelling, and understandability.
- [ ] Are there any TODOs in this PR that should be turned into stories?

### CODE REVIEW

File: pkg/tenant/api/v1/api.go [hunk 1]
```diff
@@ -251,14 +251,16 @@ type TenantProjectPage struct {
...
-	Created     string `json:"created,omitempty"`
-	Modified    string `json:"modified,omitempty"`
+	ID           string `json:"id" uri:"id"`
+	TenantID     string `json:"tenant_id"`
+	Name         string `json:"name"`
+	Description  string `json:"description,omitempty"`
+	Owner        Member `json:"owner"`
+	Status       string `json:"status"`
+	ActiveTopics uint64 `json:"active_topics"`
+	DataStorage  uint64 `json:"data_storage"`
```
Comment (pdeziel) [inline pkg/tenant/api/v1/api.go:261]: Should `data_storage` be a `StatValue`? In other words, do we need to return units to the frontend? The field is always zero for now but I'm wondering if the frontend needs to know what the units are for the list view, or we're just displaying `GB` as a hardcoded value.

File: pkg/tenant/db/projects.go [hunk 2]
```diff
@@ -21,7 +21,7 @@ const (
 
 // Project states to return to the frontend.
 const (
-	ProjectStatusIncomplete = "Incomplete Setup"
+	ProjectStatusIncomplete = "Incomplete"
```
Comment (pdeziel) [inline pkg/tenant/db/projects.go:24]: Can you check with @eschmier on if this status should be `Incomplete Setup` or `Incomplete`?

Comment (daniellemaxwell) [inline pkg/tenant/db/projects.go:24]:
```suggestion
	ProjectStatusIncomplete = "Incomplete setup"
```

**Review by daniellemaxwell — APPROVED** [review]
This looks good and the questions for clarity were greatly appreciated!

**Pull request merged**